src/poliastro/core/perturbations.py

```diff
@@ -204,6 +204,84 @@ def shadow_function(r_sat, r_sun, R):
     return theta > theta_1 + theta_2
 
 
+@jit
```

Comment:
Thanks, I have made that change. I had included it in `core/perturbations` earlier thinking it might be good to modify `radiation_pressure` accordingly once one knows whether the satellite is in sunlight, penumbra, or umbra. But we could anyways import those functions from `core/events`